nengo_loihi/builder.py

```diff
@@ -31,6 +31,250 @@
+        if self.inter_noise_exp > -30:
+            cx.enableNoise[:] = 1
+            cx.noiseExp0 = self.inter_noise_exp
+            cx.noiseAtDendOrVm = 1
+
+        syn = CxSynapses(n, label=syn_label)
+        weights2 = self.gain * np.vstack([weights, -weights] * self.n).T
+        syn.set_full_weights(weights2)
+        cx.add_synapses(syn)
+
+        return cx, syn
+
+
+class PresetInterneurons(EqualDecoderInterneurons):
+    """Abstract base class for interneurons where parameters are pre-set."""
+
+    def fix_parameters(self):
+        raise NotImplementedError()
+
+    def inter_scale(self):
+        self.fix_parameters()
+        return self.out_gain
+
+    def get_post_encoders(self, encoders):
+        encoders = encoders * self.inter_scale()
+        return np.vstack([encoders.T, -encoders.T])
+
+    def get_cx(self, weights, cx_label=None, syn_label=None):
+        self.fix_parameters()
+        d, n = weights.shape
+        cx = CxGroup(self.n * 2 * d, label=cx_label, location='core')
+        cx.configure_relu(dt=self.dt)
+        cx.bias[:] = self.bias.repeat(d)
+
+        syn = CxSynapses(n, label=syn_label)
+        weights2 = []
+        for ga, gb in self.gain.reshape(self.n, 2):
+            weights2.extend([ga*weights.T, -gb*weights.T])
+        weights2 = np.hstack(weights2)
+        syn.set_full_weights(weights2)
+        cx.add_synapses(syn)
+
+        return cx, syn
+
+
+class Preset5Interneurons(PresetInterneurons):
+    """Uses five heterogeneous interneurons with pre-set values per dimension.
+
+    The (private) script for configuring these values can be found at:
+    https://gl.appliedbrainresearch.com/abr/nengo-loihi-sandbox
+    """
+
+    def __init__(self, **kwargs):
+        super(Preset5Interneurons, self).__init__(5, **kwargs)
+
+    def fix_parameters(self):
+        assert self.n == 5
+        intercepts = np.linspace(-0.8, 0.8, self.n)
+        max_rates = np.linspace(70, 160, self.n)[::-1]
+        gain, bias = self.neuron_type.gain_bias(max_rates, intercepts)
+
+        target_point = 0.85
+        gain, bias = self.neuron_type.gain_bias(max_rates, intercepts)
+        target_rates = self.neuron_type.rates(target_point, gain, bias)
+        target_rate = target_rates.sum()
+        # TODO: why does this 1.1 factor help??
+        self.out_gain = 1.1 * target_point / (self.dt * target_rate)
+
+        self.gain = gain.repeat(2) * self.dt
+        self.bias = bias.repeat(2) * self.dt
+
+
+class Preset10Interneurons(PresetInterneurons):
+    """Uses ten heterogeneous interneurons with pre-set values per dimension.
+
+    The (private) script for configuring these values can be found at:
+    https://gl.appliedbrainresearch.com/abr/nengo-loihi-sandbox
+    """
+
+    def __init__(self, **kwargs):
+        super(Preset10Interneurons, self).__init__(10, **kwargs)
+
+    def fix_parameters(self):
+        # Parameters determined by hyperopt
+        assert self.n == 10
```

Comment:
I got rid of `fix_parameters`, since it seemed easier to just do things in the constructor. However, this is the one change that I can't do now. I don't think it's huge, but it does mean there's some overlap between the two constructors.